File: archives/2017-2018/lectures/13/Books/app/src/main/java/ro/ubbcluj/cs/books/BookApp.java
package ro.ubbcluj.cs.books;

import android.app.Application;
import android.arch.persistence.room.Room;

import ro.ubbcluj.cs.books.db.BookDatabase;
import timber.log.Timber;

public class BookApp extends Application {
  public BookDatabase db;

  @Override
  public void onCreate() {
    super.onCreate();
    Timber.plant(new Timber.DebugTree());
    db = Room.databaseBuilder(getApplicationContext(),
        BookDatabase.class, "database-name").build();
  }
}

File: lectures/13/Books/app/src/main/java/ro/ubbcluj/cs/books/BookApp.kt
package ro.ubbcluj.cs.books

import android.app.Application
import android.arch.persistence.room.Room

import ro.ubbcluj.cs.books.db.BookDatabase

class BookApp : Application() {
  lateinit var db: BookDatabase

  override fun onCreate() {
    super.onCreate()
    db = Room.databaseBuilder<BookDatabase>(applicationContext,
        BookDatabase::class.java, "database-name").build()
  }
}
